x-pack/plugins/security_solution/server/lib/detection_engine/signals/alert_suppression/group_and_bulk_create.ts

```diff
@@ -188,25 +199,33 @@ export const groupAndBulkCreate = async ({
         ruleExecutionLogger: runOpts.ruleExecutionLogger,
       });
 
-      const bulkCreateResult = await runOpts.bulkCreate(wrappedAlerts);
-
-      addToSearchAfterReturn({ current: toReturn, next: bulkCreateResult });
-
-      runOpts.ruleExecutionLogger.debug(`created ${bulkCreateResult.createdItemsCount} signals`);
-
-      const newBucketHistory: BucketHistory[] = buckets
-        .filter((bucket) => {
-          return !Object.values(bucket.key).includes(null);
-        })
-        .map((bucket) => {
-          return {
-            // This cast should be safe as we just filtered out buckets where any key has a null value.
-            key: bucket.key as Record<string, string | number>,
-            endDate: bucket.max_timestamp.value_as_string
-              ? bucket.max_timestamp.value_as_string
-              : tuple.to.toISOString(),
-          };
+      const suppressionDuration = runOpts.completeRule.ruleParams.alertSuppression?.duration;
+
+      if (suppressionDuration) {
+        const suppressionWindow = `now-${suppressionDuration.value}${suppressionDuration.unit}`;
```

Comment:
Curious about the case where a rule not configured with suppression is run, it creates some alerts, then is disabled, edited to have a suppression time period covering the previous execution, and is re-enabled. In this situation the next execution (as triggered by the re-enable) will run with suppression enabled and will suppress any new alerts that would match those alerts from the previous execution, correct?

Reply:
The suppression across rule runs depends on existing alerts having the `kibana.alert.instance.id` field populated. If the first rule execution has no suppression configured at all, then the initial alerts won't populate an instance ID. When suppression is configured, with or without a duration, the instance ID does get populated (the instance ID calculation depends on the field chosen for suppression and the value in that field, so we'd have to do a runtime computation of each alert's instance ID if the suppression field is allowed to be chosen after the alert is created).

In the scenario you described, after suppression is configured for the first time the next rule execution will still create new alerts even if the previous execution created alerts within the suppression duration with the same host name or whatever field is chosen as the suppression field.

Alternatively, if per rule execution is chosen initially and then the duration is added (without changing the suppression field) then the existing alerts from the per rule execution run would be updated, assuming they're still within the suppression duration.

Reply:
Perfect, thank you for the clarification here!